### PR TITLE
WIP: Update Pokemon attack formula

### PR DESCRIPTION
**Pull request by anapple96**
Well, this's basically the same as #583 and #585 .But, it also has Mega...at least a few of them, more will come in the future.(Megas are unobtainable, just the stats are here)

### CODE REVIEW

**Comment by anapple96** [comment]
well, technically, a few mega's stat is in the pr so...

**Comment by anapple96** [comment]
Not just an attack formula change

File: src/scripts/pokemons/PokemonList.ts [hunk 1]
```diff
@@ -13370,7 +13370,7 @@ const pokemonList: {
             'type': [PokemonType.Grass],
             'eggCycles': 20,
             'levelType': LevelType.mediumslow,
-            'exp': 230,
+            'exp': 236,
```
Comment (RedSparr0w) [inline src/scripts/pokemons/PokemonList.ts]: Could you please revert this change

File: src/scripts/pokemons/PokemonList.ts [hunk 2]
```diff
@@ -14172,12 +14172,49 @@ const pokemonList: {
...
+                'hitpoints': 78,
+                'attack': 130,
+                'specialAttack': 130,
+                'defense': 111,
+                'specialDefense': 85,
+                'speed': 100,
+            },
+
+
+        },
```
Comment (RedSparr0w) [inline src/scripts/pokemons/PokemonList.ts]: Can we also remove the mega evolutions from this PR

File: src/scripts/pokemons/PokemonList.ts [hunk 3]
```diff
@@ -14172,12 +14172,49 @@ const pokemonList: {
...
     ];
 
 
 // TODO move to its own initialize method that gets called on game start.
 pokemonList.forEach(p => {
-    p.attack = Math.max(...Object.values(p.base));
+
+    p.attack = Math.round(Math.max(Math.round(Math.max(1, Math.floor(Math.sqrt((2 * p.base.hitpoints)) * (2 * Math.round(Math.sqrt(p.base.attack * p.base.specialAttack) + Math.sqrt(p.base.speed)) * Math.sqrt(2 * Math.round(Math.sqrt(p.base.defense * p.base.specialDefense) + Math.sqrt(p.base.speed))))))) / 300));
```
Comment (RedSparr0w) [inline src/scripts/pokemons/PokemonList.ts]: Could you please reduce the complexity of this formula by splitting it up into a few extra variables.
And there are a couple redundant rounding/max/floors in here too.
I think we should also make the minimum attack be 10 as 1 is just far too low of an attack amount.

If you need any help with these just @ me on the discord chat.

File: src/scripts/pokemons/PokemonList.ts (hunk 1 above)
Comment (anapple96) [reply src/scripts/pokemons/PokemonList.ts]: fixed!

File: src/scripts/pokemons/PokemonList.ts (hunk 3 above)
Comment (anapple96) [reply src/scripts/pokemons/PokemonList.ts]: done!

File: src/scripts/pokemons/PokemonList.ts (hunk 2 above)
Comment (anapple96) [reply src/scripts/pokemons/PokemonList.ts]: gone, reduced to atoms

**Comment by anapple96** [comment]
This is merge-ready!

**Comment by anapple96** [comment]
Yeah, i think /250 is a good idea, as early game is slightly too hard with /300

**Comment by anapple96** [comment]
Hmm, /225 seems like the best one

**Comment by RedSparr0w** [comment]
Thanks for the PR 🎉
